validator/src/admin_rpc_service.rs

```diff
@@ -46,6 +48,7 @@ pub struct AdminRpcRequestMetadata {
     pub tower_storage: Arc<dyn TowerStorage>,
     pub staked_nodes_overrides: Arc<RwLock<HashMap<Pubkey, u64>>>,
     pub post_init: Arc<RwLock<Option<AdminRpcRequestMetadataPostInit>>>,
+    pub full_api: bool,
```

Comment:
I'm wondering if we even need to introduce `full_api` here for now. If the new admin endpoint was always enabled, without `--full-api` it would just be a no-op right?

Reply:
It would return success and an empty HashMap, but yeah, the overhead is not a lot -- just grabbing `meta.with_post_init` and then a Bank.

I was just thinking to parallel the handling in `solana-rpc`, but since we can't use the multiple-module approach anyway, I guess this doesn't gain us much. I'll strip it out. We can always readdress/re-add if we end up adding anything that would be more cumbersome for non-full-api nodes to support.

Reply:
I don't feel strong about this, it just seemed like a knob that eventually could be accidentally misused. Up to you!